janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/job/IndexRepairJob.java

```diff
@@ -136,21 +136,14 @@ public void process(JanusGraphVertex vertex, ScanMetrics metrics) {
 
                         Entry entry = edgeSerializer.writeRelation(janusgraphRelation, wrappedType, pos, writeTx);
                         
-                        //The below condition is check to avoid self-links which is getting created after re-indexing
-                        if(pos==0) {
-                            //Create OUT edge index entry. 
-                            //Here source will the current vertex and target vertex the other side of the relation 
-                            outAdditions.add(entry); 
-                        }else if(pos==1) {
-                            //Create IN edge index entry.
-                            //Here the source vertex is the other side of the current vertex and target will be the current vertex
+                        if (pos == 0) {
+                            outAdditions.add(entry);
+                        } else {
+                            Preconditions.checkArgument(pos == 1, "Invalid pos %d of relation", pos);
                             InternalVertex otherVertex = janusgraphRelation.getVertex(1);
                             StaticBuffer otherVertexKey = writeTx.getIdInspector().getKey(otherVertex.longId());
                             inAdditionsMap.computeIfAbsent(otherVertexKey, k -> new ArrayList<Entry>()).add(entry);
-                        }else {
-                            throw new IllegalStateException("Invalid position:" + pos);
                         }
```

Comment:
Looks like the nature of the exception is changed from IllegalStateException to IllegalArgumentException. Any reason for that?
Also, I would prefer to use direct Java code instead of Guava code where possible due to long term task #2033 . 
That said, #2033 most likely won't be resolved anytime soon because there are many places where Guava is very handy (like cache for example), where we don't have good substitutions as for now. Thus, it's OK to use Guava but if it's easy to omit it, I would prefer omitting it.
I guess we could refactor this part to use `switch` which is compiled better or same (depends on situation) than if / else chains.
Something like:
```
switch(pos){
  case 0: outAdditions.add(entry); break;
  case 1: ... break;
  default: throw new IllegalStateException("Invalid pos " + pos + " of relation");
}
```

Reply:
Removing this else branch because it is lengthy, unnecessary, and cannot be covered by the test. pos can only be 0, 1, or 2, because InternalRelation::getArity either returns 1 (for property) or 2 (for edge).

I could replace Preconditions.checkArgument with assert if that sounds better.

Reply:
`assert` sounds good to me